### DISCUSSION TITLE
UGS built-in pendant will not operate via wifi connected phone if computer is on LAN

### DISCUSSION THREAD

**Discussion by i-g-g-y** [winder/Universal-G-Code-Sender#1803] [Help]
### Version

2.0.9

### Hardware / Firmware

GRBL 1.1

### What happened

I am unable to access the built-in pendant via phone. The PC is on LAN, in this case it works flawlessin its own browser, but not on the wifi connected phone. I decided to pull the LAN out and connect PC to wifi and the pendant feature still does not work on the phone. If accessing the given url on PC it still works, but with glitches (no connection initially, but then after some time it starts).

### Operating System

WIN10

### Anything else

Occurs every time.

**Comment by i-g-g-y** [comment]
Also it issues 3 distinct IPs to connect to:
Pendant URL: http://169.254.192.27:8080Pendant URL: http://192.168.12.200:8080Pendant URL: http://192.168.56.1:8080
![image](https://user-images.githubusercontent.com/10361323/153077035-81a49ae6-8994-4f58-a1f5-fd97bc83fc89.png)
When checking URL on desktop it says not connected:
![image](https://user-images.githubusercontent.com/10361323/153077185-8886a1b4-e21e-4f63-b3dc-099ddbcbc1a7.png)
![image](https://user-images.githubusercontent.com/10361323/153077301-e7185b24-c3f5-4a4a-be58-a71c30cf3487.png)

**Comment by breiler** [comment]
First off, you are running an ancient version of UGS and you should probably upgrade.  You are not running version 2.0.9 which you specified in the issue.

The three IP-addresses means that you have three different LAN-adapters installed on your computer. And how this should work depends entirely on how you have set up your network. 

http://169.254.192.27:8080
This network adapter is probably not connected, or connected to a network without a DHCP-server and should probably be ignored

* http://192.168.12.200:8080 
* http://192.168.56.1:8080
These IP-addresses are in different series which indicates that you have two different routers running? You need to make sure that the clients on the different networks may access each other.

Do you have a firewall on the PC that restricts incoming traffic?

**Comment by i-g-g-y** [comment]
Hello, thanks for reply.

I am running 2.0.9:
![image](https://user-images.githubusercontent.com/10361323/153135383-6339a36d-930d-4eac-ad0d-bddfeeb57eab.png)

I have no idea how the 192.168.56.1 occured, as I am behind 1 router and the one is behind a master router, none of which are in the in 192.168.56.x. My router is 192.168.12.x, I opened ports in firewall and router, to allow for traffic. I also tried disabling firewall in Windows, no difference.

**Comment by i-g-g-y** [comment]
Here is the thing: I added a wireless adapter and connected to same wifi network as phone. I get now also 4th address: 192.168.12.119, and through this address it works? So it is about crossing from LAN to wifi.

![image](https://user-images.githubusercontent.com/10361323/153136202-a967eefb-9e51-4bc3-b692-0e9c305e1131.png)

**Reply by breiler** [reply]
The screenshots of the web pendant is of an ancient version. So either you have two versions running, something is cached in the browser or something is corrupt.

If I understand you correctly:
WiFi (phone) -> LAN (PC) does not work
WiFi (phone) -> WiFi (PC) does work

Check your router settings and make sure WiFi clients are allowed to communicate with LAN clients.

**Reply by i-g-g-y** [reply]
Hello, 
yes, you understand correctly. I did open ports on router and on PC firewall, this was my first suspect too. I am still going to take your advise and do another check.

For reference, this is how my UGS screen looks like:
![image](https://user-images.githubusercontent.com/10361323/153239241-df06f424-fde3-4426-b54c-8ebddc5f27b2.png)

And here is my pendant screen, if I navigate to my PC LAN address:
![image](https://user-images.githubusercontent.com/10361323/153239533-3a193584-b171-430b-9a97-36810611aafb.png)

And this is pendant screen, if I navigate on my PC wifi address: 
![image](https://user-images.githubusercontent.com/10361323/153239833-187e4b06-fd1f-4f86-86f3-15865d1970ee.png)

Both work in this case on PC, the 119 also on mobile phone on same wifi network.

**Reply by i-g-g-y** [reply]
So both work on PC, I did not touch anything.

**Comment by i-g-g-y** [comment]
So here is the thing.

About the connection issues. It was not the router where both the LAN and wifi originate.

It was the fact, that the phone jumped to another router based on signal strength. Then it stopped working.

To make matters worse, I am seeing different SW glitches, for instance the pendant stopped working even tough the phone stayed connected on correct wifi/router. I shut down UGS and restarted and it won't start server on all address, only 169 one.

Can't work with this. It is a valiant effort and I thank the community for it, but quality wise this is not something I can use.

I guess I will have to pay for professional SW, no way around this.

**Reply by breiler** [reply]
Sure, you are free to do whatever you want. 

I would probably make sure my network was working properly before opening my wallet - not sure how any software could compensate for that.

Good luck

**Reply by i-g-g-y** [reply]
True that, but the solution will circumvent the network, those solutions that feature it are even more expensive.

It is not that the network does not work, it is other glitches in UGS that make it unsuitable for steel machining. After reboot the UGS and then pendant started working again, but it is simply too much of a hassle and glitches (in movements) costs so much more in metal.

As I said, I love the principle and am grateful for all effort, but not there yet for me.

Thanks!
